### PR TITLE
Comm content retrieved from legacy bucket if fails from new bucket

### PR DESCRIPTION
**Pull request by ZsoltBalvanyos**

### CODE REVIEW

File: src/main/scala/com/ovoenergy/delivery/service/persistence/S3Repo.scala [hunk 1]
```diff
@@ -27,29 +27,35 @@ object S3Repo extends LoggingWithMDC {
...
-              case e: AmazonS3Exception => {
-                Left(
-                  AmazonS3Error(TemplateDownloadFailed,
-                                s3Config.printPdfBucketName,
-                                key.value,
-                                s"Key ${key.value} does not exist in bucket ${s3Config.printPdfBucketName}"))
+            {
+              def loadContent(bucket: String): Either[DeliveryError, Array[Byte]] = {
+                try {
+                  val document = s3Client.getObject(bucket, key.value)
```
[the remark this answers is not in the document]
Comment (laurence-bird) [reply src/main/scala/com/ovoenergy/delivery/service/persistence/S3Repo.scala]: The aws SDK doesn't support fetching from s3 via the url explicitly, you need to pass the key and the bucket and it builds the URI under the hood (no idea why this is). 

@ZsoltBalvanyos and i have just been talking about this and we're thinking for now we will have to pass the uri to extract the key and bucket for the new events (once the composer is deployed), use the key and bucket name for the legacy print comms, and use the body for the older comms.

Not ideal i know, but for the interim seems like the best approach until we do the backfill, and then we can come back and refactor to use the comms-aws library. Thoughts? @filosganga @SystemFw

[the remark this answers is not in the document]
Comment (filosganga) [reply src/main/scala/com/ovoenergy/delivery/service/persistence/S3Repo.scala]: It is fine, I was just wondering if you were aware that we don't have the key explicitly and we need to extract from the URI.

File: src/main/scala/com/ovoenergy/delivery/service/persistence/CommContent.scala [hunk 2]
```diff
@@ -29,40 +31,48 @@ object CommContent {
...
-        .map(Content.Print)
-  }
+      override def getPrintContent(composedPrintV2: ComposedPrintV2): F[Content.Print] =
+        if (isUrl(composedPrintV2.pdfIdentifier)) {
+          val s3Uri = new AmazonS3URI(composedPrintV2.pdfIdentifier)
+          s3Repo
+            .getDocument(Key(s3Uri.getKey), Bucket(s3Uri.getBucket))
+            .map(Content.Print)
+        } else
+          s3Repo
```
Comment (laurence-bird) [inline src/main/scala/com/ovoenergy/delivery/service/persistence/CommContent.scala]: It would be good to verify this behaviour in a unit test i think

**Comment by ovo-comms-circleci** [comment]
This PR was successfully deployed to production by [CircleCI build 470](https://circleci.com/gh/ovotech/comms-delivery-service/470) at 2018-10-02 16:58 BST.

**Pull request merged**